crates/relayer/src/chain/axon.rs

```diff
@@ -687,6 +701,83 @@ impl ChainEndpoint for AxonChain {
     ) -> Result<Vec<CrossChainQueryResponse>, Error> {
         todo!()
     }
+
+    fn build_connection_proofs_and_client_state(
```

Comment:
most of the code here is the same as the channel and packet operation, so consider wrapping it as a common method to simplify the further use